### ISSUE TITLE
Binary not executing properly on Mac OSX

### ISSUE THREAD

**Issue by imdyangs** [minimalchat/daemon#35]
### Problem Description:

If you try to execute the [binary](https://github.com/minimalchat/daemon/releases) on Mac OSX, it errors and spits back:

```
exec format error: ./daemon
```

### Suggested Solution:
Build a binary on Mac that works and add a separate release.

**Comment by nynhex** [comment]
@imdyangs have you tried actually doing a Go build?  I built on high sierra

**Comment by mihok** [comment]
Yeah good call @nynhex and @imdyangs, Mac OSX builds built on a linux box may not work as expected... Unfortunately I don't have a Mac OSX machine to leverage so someone else will have to make a proper build to include in the release for Mac users

**Comment by nynhex** [comment]
@mihok I have a High Sierra/Mojave Machine.  I can do some proper builds and PR if you like.  cc: @imdyangs

**Comment by mihok** [comment]
That would be very appreciated @nynhex !

**Comment by EyePulp** [comment]
I'm new to Go, so this is a dumb one, but how do I build this on OSX? I have go 1.11x installed, but I'm not sure if I should be using the Makefile or not.  The makefile seems to have some hardcoded paths and I don't know if I should be editing those or changing my environment to match them. Suggestions appreciated. =)

**Comment by mihok** [comment]
@EyePulp thanks for the interest in Minimal Chat!!

You _should_ be able to use the Makefile provided that your go environment is set up properly.  Go has a very opinionated folder structure it follows ([see here](https://golang.org/doc/code.html#Organization)) have you set your $GOPATH env var?

Otherwise, take a look at the `compile:` command on the Makefile, it will illustrate how you might build the daemon without the Makefile

**Comment by EyePulp** [comment]
@mihok thanks much for the guidance.  Learning about the folder layout now - sorry for not asking in a separate issue, I realize this is the wrong spot.
